core/test/XiosFile_test.cpp

```diff
@@ -73,19 +76,46 @@ MPI_TEST_CASE("TestXiosFile", 2)
     xios_handler.setFileType(fileId, fileType);
     REQUIRE(xios_handler.getFileType(fileId) == fileType);
     // Output frequency
-    const std::string freq { "1ts" };
-    xios_handler.setFileOutputFreq(fileId, freq);
-    REQUIRE(xios_handler.getFileOutputFreq(fileId) == freq);
+    xios_handler.setFileOutputFreq(fileId, timestep);
+    REQUIRE(xios_handler.getFileOutputFreq(fileId).seconds() == Approx(1.5 * 60 * 60));
     // Split frequency
-    const std::string splitFreq { "1ts" };
-    xios_handler.setFileSplitFreq(fileId, splitFreq);
-    REQUIRE(xios_handler.getFileSplitFreq(fileId) == splitFreq);
+    xios_handler.setFileSplitFreq(fileId, timestep);
+    REQUIRE(xios_handler.getFileSplitFreq(fileId).seconds() == Approx(1.5 * 60 * 60));
     // Add field
     xios_handler.fileAddField(fileId, "field_A");
     std::vector<std::string> fieldIds = xios_handler.fileGetFieldIds(fileId);
     REQUIRE(fieldIds.size() == 1);
     REQUIRE(fieldIds[0] == "field_A");
 
+    // Create a new file for each time unit to check more thoroughly that XIOS interprets output
+    // frequency and split frequency correctly.
+    // (If we reused the same file then the XIOS interface would raise warnings.)
+    xios_handler.createFile("year");
+    xios_handler.setFileOutputFreq("year", Duration("P1-0T00:00:00"));
+    xios_handler.setFileSplitFreq("year", Duration("P2-0T00:00:00"));
+    REQUIRE(xios_handler.getFileOutputFreq("year").seconds() == Approx(365 * 24 * 60 * 60));
+    REQUIRE(xios_handler.getFileSplitFreq("year").seconds() == Approx(2 * 365 * 24 * 60 * 60));
+    xios_handler.createFile("day");
+    xios_handler.setFileOutputFreq("day", Duration("P0-1T00:00:00"));
+    xios_handler.setFileSplitFreq("day", Duration("P0-2T00:00:00"));
+    REQUIRE(xios_handler.getFileOutputFreq("day").seconds() == Approx(24 * 60 * 60));
+    REQUIRE(xios_handler.getFileSplitFreq("day").seconds() == Approx(2 * 24 * 60 * 60));
+    xios_handler.createFile("hour");
+    xios_handler.setFileOutputFreq("hour", Duration("P0-0T01:00:00"));
+    xios_handler.setFileSplitFreq("hour", Duration("P0-0T02:00:00"));
+    REQUIRE(xios_handler.getFileOutputFreq("hour").seconds() == Approx(60 * 60));
+    REQUIRE(xios_handler.getFileSplitFreq("hour").seconds() == Approx(2 * 60 * 60));
+    xios_handler.createFile("minute");
+    xios_handler.setFileOutputFreq("minute", Duration("P0-0T00:01:00"));
+    xios_handler.setFileSplitFreq("minute", Duration("P0-0T00:02:00"));
+    REQUIRE(xios_handler.getFileOutputFreq("minute").seconds() == Approx(60));
+    REQUIRE(xios_handler.getFileSplitFreq("minute").seconds() == Approx(2 * 60));
+    xios_handler.createFile("second");
+    xios_handler.setFileOutputFreq("second", Duration("P0-0T00:00:01"));
+    xios_handler.setFileSplitFreq("second", Duration("P0-0T00:00:02"));
+    REQUIRE(xios_handler.getFileOutputFreq("second").seconds() == Approx(1));
+    REQUIRE(xios_handler.getFileSplitFreq("second").seconds() == Approx(2));
+
```

Comment:
I know that comparing floating point values can be tricky, but in this case, the values should be exact. If the duration is being set to exactly one second, and the returned floating point value is not the exact value 1, this should also be a failure, and so on for all the othre exact integer values of seconds.

Reply:
Thanks @timspainNERSC, good point. This should be fixed by 71ff796.